docs/cookbook/cli/handle-multiple-channels-in-cli.rst

```diff
@@ -0,0 +1,170 @@
+Handle multiple channels in CLI
+===============================
+
+When we use directly or indirectly any service depending on a channel context, we are not able to define which channel we want to use, when there are more than two.
+Your primary goal should be to avoid such cases, but if you have to, there is a way to do it.
+
+What is our goal?
+-----------------
+
+To accomplish this task, we have to create a custom channel context available only from the CLI context. But this concrete channel context cannot only return a channel,
+but also has to allow to pass a channel code as an argument (which we will get from the console command). This way, we can use the channel context in our services.
+
+1. Custom Channel Context
+-------------------------
+
+First, we need to create a custom channel context. Let's remind our requirements:
+
+    * available only from the CLI
+    * there must be a way to pass in a channel code
+
+Our suggested solution is the following:
+
+.. code-block:: php
+
+    // src/Channel/Context/CliBasedChannelContext.php
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Channel\Context;
+
+    use Sylius\Component\Channel\Context\ChannelNotFoundException;
+    use Sylius\Component\Channel\Model\ChannelInterface;
+    use Sylius\Component\Channel\Repository\ChannelRepositoryInterface;
+
+    final class CliBasedChannelContext implements CliBasedChannelContextInterface
+    {
+        private ?string $channelCode = null;
+
+        public function __construct(
+            private ChannelRepositoryInterface $channelRepository,
+        ) {
+        }
+
+        public function setChannelCode(?string $channelCode): void
+        {
+            $this->channelCode = $channelCode;
+        }
+
+        public function getChannelCode(): ?string
+        {
+            return $this->channelCode;
+        }
+
+        public function getChannel(): ChannelInterface
+        {
+            if ('cli' !== PHP_SAPI || null === $this->channelCode) {
+                throw new ChannelNotFoundException();
+            }
+
+            $channel = $this->channelRepository->findOneByCode($this->channelCode);
+
+            if (null === $channel) {
+                throw new ChannelNotFoundException();
+            }
+
+            return $channel;
+        }
+    }
+
+.. code-block:: php
+
+    // src/Channel/Context/CliBasedChannelContextInterface.php;
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Channel\Context;
+
+    use Sylius\Component\Channel\Context\ChannelContextInterface;
+
+    interface CliBasedChannelContextInterface extends ChannelContextInterface
+    {
+        public function setChannelCode(?string $channelCode): void;
+
+        public function getChannelCode(): ?string;
```

Comment:
It can be used to check whether a channel code exists to prevent overriding. And still, it's just an example which might be adjusted to the concrete case.